### ISSUE TITLE
Add derive macro for View

### ISSUE THREAD

**Issue by ecton** [khonsulabs/bonsaidb#139]
* ~~Blocked by #99~~

Once we split View implementations from their metadata, View could be derived in a similar fashion as #138.

The macro needs to be able to have the location of `bonsaidb_core` specified as an optional attribute. By default, it should use `::bonsaidb::core` as the path to bonsaidb_core, but a user should be able to override it based on their usage. E.g., `#[view(core = bonsaidb_server::core)]`.

**Comment by ModProg** [comment]
I you propose some attributes and implementation details, I'll implement the macro.

**Comment by ecton** [comment]
Yes, please!

**Comment by ModProg** [comment]
https://github.com/khonsulabs/bonsaidb/blob/aad865032b54e3e095dfb087884be134e772f683/crates/bonsaidb-core/src/schema/view.rs#L60-L126

It seems that version is in `ViewSchema`, but I don't think you would be able to derive the `map` function?

**Comment by ecton** [comment]
You're right, I typed that example from memory and I forgot what was where. Version is indeed for the schema portion of the trait.

ViewSchema I don't think will be derivable. We may want some sort of DSL-like macro to cut down some boilerplate, but I'll put in another issue to ponder that idea.

**Comment by ModProg** [comment]
So what is the state here, should this be closed? did you create a new issue for:
> ViewSchema I don't think will be derivable. We may want some sort of DSL-like macro to cut down some boilerplate, but I'll put in another issue to ponder that idea.

**Comment by ecton** [comment]
I'm going to go ahead and consider this fully resolved -- the other tickets can handle the other requests for new/updated macros as needed.
